nussl/separation/duet.py

```diff
@@ -113,102 +164,78 @@ def run(self):
         time_vector = self.audio_signal.time_bins_vector
 
         # remove dc component to avoid dividing by zero freq. in the delay estimation
-        stft_ch0 = stft_ch0[1::, :]
-        stft_ch1 = stft_ch1[1::, :]
-        num_frequency_bins = len(frequency_vector)
-        num_time_bins = len(time_vector)
+        stft_ch0 = stft_ch0[1:, :]
+        stft_ch1 = stft_ch1[1:, :]
+        self.num_frequency_bins = len(frequency_vector)
+        self.num_time_bins = len(time_vector)
 
         # Compute the freq. matrix for later use in phase calculations
-        frequency_matrix = np.array(np.tile(np.mat(frequency_vector[1::]).T, (1, num_time_bins))) * (2 * np.pi / fs)  # WTF?
+        wmat = np.array(np.tile(np.mat(frequency_vector[1:]).T, (1, self.num_time_bins))) * (2 * np.pi / sample_rate)
+
+        return stft_ch0, stft_ch1, wmat
 
+    def _compute_atn_delay(self):
         # Calculate the symmetric attenuation (alpha) and delay (delta) for each
         # time-freq. point
-        R21 = (stft_ch1 + nussl.constants.EPSILON) / (stft_ch0 + nussl.constants.EPSILON)
-        relative_attenuation = np.abs(R21)  # relative attenuation between the two channels
-        symmetric_attenuation = relative_attenuation - 1 / relative_attenuation
-        relative_delay = -np.imag(np.log(R21)) / (2 * np.pi * frequency_matrix)  # relative delay
+        inter_channel_ratio = (self.stft_ch1 + nussl.constants.EPSILON) / (self.stft_ch0 + nussl.constants.EPSILON)
+        atn = np.abs(inter_channel_ratio)  # relative attenuation between the two channels
+        symmetric_atn = atn - 1 / atn  # symmetric attenuation
+        delay = -np.imag(np.log(inter_channel_ratio)) / (2 * np.pi * self.frequency_matrix)  # relative delay
+        return symmetric_atn, delay
+
+    def make_histogram(self, p, q):
```

Comment:
Does this function need to be public? Would a user use this function by itself, or does it only work in the context of the rest of the code? If it's the latter, then make it private.